### PR TITLE
Using rocm-cmake for client packaging

### PR DESCRIPTION
**Pull request by saadrahim**

### CODE REVIEW

File: clients/CMakeLists.txt [hunk 1]
```diff
@@ -116,13 +116,9 @@ add_custom_command(OUTPUT "${ROCSPARSE_GENTEST}"
...
-include(../cmake/Packaging.cmake)
-
 if(BUILD_CLIENTS_SAMPLES OR BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
-  rocm_create_package_clients(LIB_NAME rocsparse
-                              DESCRIPTION "Radeon Open Compute SPARSE library"
-                              MAINTAINER "rocSPARSE Maintainer <rocsparse-maintainer@amd.com>"
-                              SECTION "dev"
-                              VERSION ${rocsparse_VERSION})
+        install(
+                DIRECTORY "${CMAKE_SOURCE_DIR}/build/release/clients/staging/"
```
Comment (YvanMokwinski) [inline clients/CMakeLists.txt:121]: we can't hard code the 'release' directory

Comment (YvanMokwinski) [reply clients/CMakeLists.txt:121]: use of CMAKE_BINARY_DIR instead ? :
"${CMAKE_BINARY_DIR}/clients/staging/"

**Review by YvanMokwinski — CHANGES_REQUESTED** [review]

**Comment by YvanMokwinski** [comment]
Other question, why don't we create this PR on rocSPARSE-internal?

**Comment by saadrahim** [comment]
This PR was from a public fork. The content of the work is not mine.

File: clients/CMakeLists.txt (hunk 1 above)
Comment (TorreZuk) [reply clients/CMakeLists.txt:121]: Or "${PROJECT_BINARY_DIR}/staging/*"  is what rocblas uses.  We  completely override rocm_create_package_clients not sure if the original never worked or who did the override.  See clients/CMakeLists.txt in rocblas for alternative example, @daineAMD any comment on this?

Comment (daineAMD) [reply clients/CMakeLists.txt:121]: Looks like Saad is the one the originally wrote client packaging (see 
https://github.com/ROCmSoftwarePlatform/rocBLAS/pull/602)

It looks to me like earlier in this file ${PROJECT_BINARY_DIR}/staging is used in the same sense as here (and in rocBLAS).

**Comment by ntrost57** [comment]
It fails on Ubuntu when trying to generate the client packages:
```+ make package_clients
make: *** No rule to make target 'package_clients'.  Stop.
+ mkdir -p package
+ mv rocsparse_1.20.2-220840d_amd64.deb package/
+ mv 'clients/*.deb' package/
mv: cannot stat 'clients/*.deb': No such file or directory
script returned exit code 1

**Comment by saadrahim** [comment]
New PR coming from @lawruble13 to replace this in future.

**Pull request closed without merging**